From adc38f902a8d2066b65df571aa21ed7de2a99b46 Mon Sep 17 00:00:00 2001
From: Robert Mader <robert.mader@posteo.de>
Date: Tue, 25 Feb 2020 00:14:40 +0100
Subject: [PATCH] window-actor/X11: Update shape, input and opaque region in
 order

As they depend on each other to be correct, we should set all of them
in the correct order. As we do already have a function for that, use it.

https://gitlab.gnome.org/GNOME/mutter/merge_requests/1091
---
 src/compositor/meta-window-actor-x11.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/compositor/meta-window-actor-x11.c b/src/compositor/meta-window-actor-x11.c
index ef50b1ada..04b0ab5c5 100644
--- a/src/compositor/meta-window-actor-x11.c
+++ b/src/compositor/meta-window-actor-x11.c
@@ -1138,10 +1138,7 @@ handle_updates (MetaWindowActorX11 *actor_x11)
        * which causes the shadows to look bad.
        */
       if (surface && meta_window_x11_always_update_shape (window))
-        {
-          update_opaque_region (actor_x11);
-          update_shape_region (actor_x11);
-        }
+        check_needs_reshape (actor_x11);
 
       return;
     }
-- 
2.24.1

